externs/browser/chrome.js

```diff
@@ -76,29 +76,170 @@ function ChromeEvent() {}
 /**
  * @param {!Function} callback
  * @return {undefined}
+ * @see https://developer.chrome.com/extensions/events#method-Event-addListener
  */
 ChromeEvent.prototype.addListener = function(callback) {};
 
 
 /**
  * @param {!Function} callback
  * @return {undefined}
+ * @see https://developer.chrome.com/extensions/events#method-Event-removeListener
  */
 ChromeEvent.prototype.removeListener = function(callback) {};
 
 
 /**
  * @param {!Function} callback
  * @return {boolean}
+ * @see https://developer.chrome.com/extensions/events#method-Event-hasListener
  */
 ChromeEvent.prototype.hasListener = function(callback) {};
 
 
-/** @return {boolean} */
+/**
+ * @return {boolean}
+ * @see https://developer.chrome.com/extensions/events#method-Event-hasListeners
+ */
 ChromeEvent.prototype.hasListeners = function() {};
 
-// TODO(tbreisacher): Add the addRules, getRules, and removeRules methods?
 
+/**
+ * @param {!Array<Rule>} rules
+ * @param {function(!Array<Rule>): void=} callback
```

Comment:
Are you allowed to have null rules in the array?
If not, please make this `!Array<!Rule>`.
If so, please make nullability explicit with `!Array<?Rule>`.
Same for all other similar cases.